storage/imxrt-flash/flashsrv.c

```diff
@@ -23,18 +23,21 @@
 #include <meterfs.h>
 #include <ptable.h>
 
+#include "fspi.h"
 #include "flashdrv.h"
 #include "imxrt-flashsrv.h"
 
 
 #define LOG_ERROR(str, ...) do { fprintf(stderr, __FILE__  ":%d error: " str "\n", __LINE__, ##__VA_ARGS__); } while (0)
 #define TRACE(str, ...) do { if (0) fprintf(stderr, __FILE__  ":%d trace: " str "\n", __LINE__, ##__VA_ARGS__); } while (0)
 
-#define METERFS_STACKSZ              1024
-#define THREAD_STACKSZ               512
-#define THREAD_PRIORITY              4
+#define METERFS_STACKSZ 1024
+#define THREAD_STACKSZ  512
 
-#define FLASH_MEMORIES_NO            2
+/* FIXME: temporarily highest priority is used to avoid preemption of programs that use high priorities and run from XIP. */
+#define THREAD_PRIORITY 0
```

Comment:
Also, this is a very good candidate for board_config.h, as this parameter is quite essential for a good system configuration.

Reply:
Yes, we already agreed that `THREAD_PRIORITY` will be `3`, one below default `4`. First we need to review priorities in other apps which register irq's, which are executed in ram priority (0-3)  and which from flash/xip (4-7). Hence it is `FIXME:` marked. I think yet before merging this it will be possible to set priority level to `3`.

Reply:
Added.